tests/lang_callable/macros/tmacros_various.nim

```diff
@@ -328,3 +330,16 @@ block: # bug #15118
 
   block:
     flop("b")
+
+block: # Ensure nkCommentStmt equality is not ignored when checking for const duplicates:
+  proc createComment(s: string): NimNode =
+    result = nnkCommentStmt.newNimNode()
+    result.strVal = s
+
+  macro testComments =
+    const C1 = (1, createComment("comment 1"))
+    const C2 = (1, createComment("comment 2"))
+    echo treeRepr(C1[1])
+    echo treeRepr(C2[1])
+
+  testComments
```

Comment:
```suggestion
  const C1 = (1, createComment("comment 1"))
  const C2 = (1, createComment("comment 2"))
  static:
    echo treeRepr(C1[1])
    echo treeRepr(C2[1])
```
A soft suggestion for reducing the number of used language features a bit (macro expansion + compile-time evaluations vs. only compile-time evaluation).